lttoolbox/att_compiler.cc

```diff
@@ -52,21 +53,24 @@ AttCompiler::clear()
  *       for conversion?
  */
 void
-AttCompiler::convert_hfst(wstring& symbol)
+AttCompiler::convert_hfst(UString& symbol)
 {
-  if (symbol == L"@0@" || symbol == L"ε")
-  {
-    symbol = L"";
-  }
-  else if (symbol == L"@_SPACE_@")
-  {
-    symbol = L" ";
+  if (symbol == Transducer::HFST_EPSILON_SYMBOL_SHORT ||
+      symbol == Transducer::HFST_EPSILON_SYMBOL_LONG ||
+      symbol == Transducer::LTTB_EPSILON_SYMBOL) {
+    symbol.clear();
+  } else if (symbol == Transducer::HFST_SPACE_SYMBOL) {
+    symbol = " "_u;
+  } else if (symbol == Transducer::HFST_TAB_SYMBOL) {
+    symbol = "\t"_u;
   }
 }
 
 bool
-AttCompiler::is_word_punct(wchar_t symbol)
+AttCompiler::is_word_punct(UChar32 symbol)
 {
+  // this version isn't quite write, but something like it should be possible
```

Comment:
I assume "write" was on purpose

Reply:
nope, just further evidence that the phonetic and orthographic forms are separate in my brain :)